### PR TITLE
Makefile,snapcraft: fix the copy of buildd sources, do not setup FIPS PPA on LP

### PR DESCRIPTION
**Pull request by Meulengracht**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,9 +29,9 @@ install:
 	# ensure resolving works inside the chroot
 	cat /etc/resolv.conf > $(DESTDIR)/etc/resolv.conf
 	# copy-in launchpad's build archive
-	if grep -q ftpmaster.internal /etc/apt/sources.list; then \
+	if grep -q ftpmaster.internal /etc/apt/sources.list.d/lp-buildd.sources; then \
```
Comment (Meulengracht) [inline Makefile:32]: From Noble (24.04) buildd no longer puts sources into `sources.list` but instead truncates this file and writes `/etc/apt/sources.list.d/lp-buildd.sources`

File: Makefile [hunk 2]
```diff
@@ -28,9 +29,9 @@ install:
...
 		cp /etc/apt/sources.list $(DESTDIR)/etc/apt/sources.list; \
-		cp /etc/apt/trusted.gpg $(DESTDIR)/etc/apt/ || true; \
```
Comment (Meulengracht) [inline Makefile:33]: This file no longer exists

File: Makefile [hunk 3]
```diff
@@ -28,9 +29,9 @@ install:
...
+		cp /etc/apt/sources.list.d/lp-buildd.sources $(DESTDIR)/etc/apt/sources.list.d/lp-buildd.sources; \
```
Comment (bboozzoo) [inline Makefile:34]: does this need to be conditional?

File: Makefile [hunk 4]
```diff
@@ -10,6 +10,7 @@ CODENAME:="$(shell . /etc/os-release; echo "$$VERSION_CODENAME")"
 -include .fips-env
 ifdef SNAP_FIPS_BUILD
     export SNAP_FIPS_BUILD
+    export SNAP_BUILD_NAME
```
Comment (bboozzoo) [inline Makefile:13]: why not always do the export? and if not, the comment right before `-include` needs to be updated

File: Makefile (hunk 3 above)
Comment (Meulengracht) [reply Makefile:34]: should probably match the existing pattern yes

**Review by bboozzoo — APPROVED** [review]

**Review by valentindavid — APPROVED** [review]

**Pull request merged**